packages/common/src/http.js

```diff
@@ -0,0 +1,152 @@
+ * @function
+ * @public
+ * @example <caption>Get with a query an oath token</caption>
+ * get($.data.url, http.options({ query: $.query }).oath($.configuration.access_token)
+ */
+export function options(opts = {}) {
+  for (let h in helpers) {
+    Object.defineProperty(opts, h, {
+      enumerable: false,
+      value: helpers[h],
+    });
+  }
+
+  return opts;
+}
+
+/**
+ * Options provided to the HTTP request
+ * @typedef {Object} CommonRequestOptions
+ * @property {object} errors - Map of errorCodes -> error messages, ie, `{ 404: 'Resource not found;' }`. Pass `false` to suppress errors.
+ * @property {object} form - Pass a JSON object to be serialised into a multipart HTML form (as FormData) in the body.
+ * @property {object} query - An object of query parameters to be encoded into the URL.
+ * @property {object} headers - An object of headers to append to the request.
+ * @property {string} parseAs - Parse the response body as json, text or stream. By default will use the response headers.
+ * @property {number} timeout - Request timeout in ms. Default: 300 seconds.
+ * @property {object} tls - TLS/SSL authentication options. See https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions
+ */
+
+/**
+ * State object
+ * @typedef {Object} CommonHttpState
+ * @private
+ * @property data - the parsed response body
+ * @property response - the response from the HTTP server, including headers, statusCode, body, etc
+ * @property references - an array of all previous data objects used in the Job
+ **/
+
+/**
+ * Make a HTTP request.
+ * @public
+ * @function
+ * @example
+ * http.request(
+ *   'GET',
+ *   'https://jsonplaceholder.typicode.com/todos'
+ * )
+ * @name request
+ * @param {string} method - The HTTP method to use.
+ * @param {string} url - URL to resource.
+ * @param {CommonRequestOptions} options - Request options
+ * @state {CommonHttpState}
+ * @returns {Operation}
+ */
+const req = function (method, url, options) {
+  return state => {
+    const [resolvedMethod, resolvedUrl, resolvedOptions] = expandReferences(
+      state,
+      method,
+      url,
+      options
+    );
+    return request(resolvedMethod, resolvedUrl, resolvedOptions);
+  };
+};
+export { req as request };
+
+/**
+ * Make a GET request.
+ * @public
+ * @function
+ * @example <caption>Request a resource</caption>
+ * http.get('https://jsonplaceholder.typicode.com/todos')
+ * @example <caption>Request a resource with basic auth</caption>
+ * http.get(
+ *  'https://jsonplaceholder.typicode.com/todos',
+ *  http.options().basic('user', 'pass')
+ * )
+ * @example <caption>Request a resource with oauth</caption>
+ * http.get(
+ *  'https://jsonplaceholder.typicode.com/todos',
+ *  http.options().oauth($.configuration.access_token)
+ * )
+ * @param {string} url - URL to access
+ * @param {CommonRequestOptions} options - Request options
+ * @state {CommonHttpState}
+ * @returns {Operation}
+ */
+export function get(url, options) {
```

Comment:
I have tested this in `asana` it works but it overwrites `state`

Reply:
Oh crap, you're right! I've done it wrong!

Reply:
That should be fixed now, thank you :pray:

Reply:
This is working as expected now 👍🏽